### PR TITLE
Annotate `entity.lua`

### PR DESCRIPTION
**Pull request by Garanas**
This branch is for bugfixes, general improvements, and new features. If your change is designed to alter balance, please make 
sure your changes are rebased onto [development/balance] (https://github.com/FAForever/fa/tree/development/balance), and target that branch with your PR.

### CODE REVIEW

File: engine/Sim/Entity.lua [hunk 1]
```diff
@@ -289,7 +324,11 @@ end
 
 ---
 --  Entity:SetPosition(vector,[immediate])
-function Entity:SetPosition(vector, [immediate])
+
+---@see Warp
+---@param vector Position
+---@param immediate boolean Defaults to false, should not be required
+function Entity:SetPosition(vector, )
```
Comment (Hdt80bro) [inline engine/Sim/Entity.lua:331]: You dropped `immediate`!

File: engine/Sim/Entity.lua [hunk 2]
```diff
@@ -177,14 +192,18 @@ end
...
+--- Returns the position of the entity at a given bone, or at its center bone as a table that is refreshed on each call
+---@param bone? string | number
+---@return Vector
+function Entity:GetPosition(bone)
 end
 
----
---  Entity:GetPositionXYZ([bone_name])
-function Entity:GetPositionXYZ([bone_name])
+--- Returns the position of the entity at a given bone, or at its center bone as three separate numbers
```
Comment (Hdt80bro) [inline engine/Sim/Entity.lua:201]: A little confusing wording
Maybe `Returns the position of the entity at a given bone (or its center bone if absent) as three separate numbers`? You could offset the parenthetical with commas instead, I don't mind.

File: engine/Sim/Entity.lua [hunk 3]
```diff
@@ -5,8 +5,10 @@ local Entity = {}
 ---@alias Army number
 ---@alias EntityId number
 
----
---  Entity:AddManualScroller(scrollSpeed1, scrollSpeed2)
+--- Does not appear to be used
+---@unknown
+---@param scrollSpeed1 number
+---@param scrollSpeed2 number
 function Entity:AddManualScroller(scrollSpeed1,  scrollSpeed2)
```
Comment (Hdt80bro) [inline engine/Sim/Entity.lua:12]: You   left   in   all   of   the   extra   spaces   separating   parameters   in   every   function (though it doesn't really matter)

File: engine/Sim/Entity.lua [hunk 4]
```diff
@@ -204,7 +223,12 @@ end
 
 ---
 --  Entity:IsValidBone(nameOrIndex,allowNil=false)
-function Entity:IsValidBone(nameOrIndex, allowNil=false)
+
+--- Returns whether the bone is a valid bone for the entity
+---@param bone number | string
+---@param allowNil boolean Flag that to consider nil a valid bone, defaults to false
```
Comment (Hdt80bro) [inline engine/Sim/Entity.lua:229]: If it has a default value, make the parameter optional

File: engine/Sim/Entity.lua [hunk 5]
```diff
@@ -249,7 +273,14 @@ end
 
 ---
 --  Entity:SetCollisionShape(['Box'|'Sphere'|'None'], centerX, Y, Z, size) -- size is radius for sphere, x,y,z extent for box
-function Entity:SetCollisionShape(['Box'|'Sphere'|'None'],  centerX,  Y,  Z,  size)
+
+--- Defines the collision shape of the entity. Should not be used excessively due to its performance impact
+---@param type 'Box' | 'Sphere' | 'None' 
+---@param centerX number
+---@param Y number
+---@param Z number
```
Comment (Hdt80bro) [inline engine/Sim/Entity.lua:281]: Are these not named `center` for a reason?

Comment (Garanas) [reply engine/Sim/Entity.lua:296]: It is a function with a strange overload 😄 

It is either:

- a box (type, X, Y, Z, nil)
- A sphere (type, center, nil, nil, radius)

**Comment by Garanas** [comment]
I'm merging this in favor for someone else to continue it, I don't have the time to program that much these days 😄

File: engine/Sim/Entity.lua (hunk 5 above)
Comment (Hdt80bro) [reply engine/Sim/Entity.lua:298]: That's... so fun

**Comment by Hdt80bro** [comment]
Sure, I'll pick it up

**Pull request merged**